pycodestyle.py

```diff
@@ -134,8 +134,10 @@
                                      r'\s*(?(1)|(None|False|True))\b')
 COMPARE_NEGATIVE_REGEX = re.compile(r'\b(?<!is\s)(not)\s+[^][)(}{ ]+\s+'
                                     r'(in|is)\s')
-COMPARE_TYPE_REGEX = re.compile(r'(?:[=!]=|is(?:\s+not)?)\s+type(?:s.\w+Type'
```

Comment:
I can't find the rationale for removing `types.*Type` comparisons in the linked issues. As far as I can tell, those should still be included in the checks, shouldn't they?

Reply:
I think those got removed because the name `types` isn't necessarily the stdlib `types` module -- not 100% sure though

Reply:
the original patch is here: https://github.com/PyCQA/pycodestyle/pull/1041 -- this is just a revert-revert of that with an additional commit on top

Reply:
I suppose the precise reasoning is the following, then:

> This check also breaks many ligimitate cases (such as having your own types module with an Enum class for instance, and comparing a value to the enum member).

which I can relate to. You can, of course, also shadow the built-in `type` function, but that's just not something people do in general and having a custom `types` module is much more likely, although also ending your enum in `Type` is most likely less common.